src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerServerCredentialsBinding.java

```diff
@@ -0,0 +1,59 @@
+package org.jenkinsci.plugins.docker.commons.credentials;
+
+import java.io.IOException;
+
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.credentialsbinding.BindingDescriptor;
+import org.jenkinsci.plugins.credentialsbinding.impl.AbstractOnDiskBinding;
+import org.jenkinsci.plugins.docker.commons.Messages;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import hudson.Extension;
+import hudson.FilePath;
+
+public class DockerServerCredentialsBinding extends AbstractOnDiskBinding<DockerServerCredentials> {
+
+    @DataBoundConstructor
+    public DockerServerCredentialsBinding(String variable, String credentialsId) {
+        super(variable, credentialsId);
```

Comment:
Would be possible to extend `MultiBinding` directly (using `UnbindableDir`), which would enable you to supply a default value for `variable` by turning it into a `@DataBoundSetter`. (In the current inheritance chain this is impossible since the field is `final`.) Would enable Pipeline syntax to be simply

```groovy
withCredentials([dockerCert(credentialsId: 'docker-client-cert')]) {
  sh 'docker push'
}
```

Not sure if it is worth the effort to make that refactoring, just FYI.